api/app/attendance/api.py

```diff
@@ -115,16 +109,31 @@ def get(self):
             confirmed = registration.confirmed if registration is not None else True
             invitedguest_role = "General Attendee"
             is_invited_guest = False
-        # Shirt Size
-        shirt_answer = _get_registration_answer(user_id, event_id, "T-Shirt Size", is_invited_guest)
-
-        # Poster registration
-        bringing_poster = _get_registration_answer(user_id, event_id, "Will you be bringing a poster?", is_invited_guest) == 'yes'
-
-        attendance_user = AttendanceUser(user, attendance, accommodation_award=has_accepted_accom_award, 
-                                         shirt_size=shirt_answer, is_invitedguest=is_invited_guest, 
-                                         bringing_poster=bringing_poster, invitedguest_role=invitedguest_role,
-                                         confirmed=confirmed)
+        
+        # collate all tags belonging to user into a list of (tag_id, tag_name, answer) tuples 
+        # first, get all tags from registration questions
+        registration_metadata = []
+        questions_with_tags = registration_form_repository.get_registration_questions_with_tags(event_id)
+        for question in questions_with_tags:
+            for question_tag in question.tags:
+                answer = _get_registration_answer(user_id, event_id, question.id, is_invited_guest)
+                registration_metadata.append({"name": TagAPI._stringify_tag_name(question_tag.tag), "response": answer})
+        print(registration_metadata)
+
+        # second, get all tags from offers
+        offer_metadata = []
+        offer = db.session.query(Offer).filter(
+            Offer.user_id == user_id).filter(Offer.event_id == event_id).first()
+        if offer:
+            offer_tags = db.session.query(OfferTag).filter(

```

Comment:
You can just directly use the `.offer_tags` property on the offer object instead of a new query.